bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/UniqueConfig.scala

```diff
@@ -0,0 +1,112 @@
+package jsky.app.ot.editor.seq
+
+import edu.gemini.spModel.config2.{Config, ConfigSequence}
+import edu.gemini.spModel.gemini.acqcam.InstAcqCam
+import edu.gemini.spModel.gemini.flamingos2.Flamingos2
+import edu.gemini.spModel.gemini.gmos.{GmosNorthType, GmosSouthType, InstGmosNorth, InstGmosSouth}
+import edu.gemini.spModel.gemini.gnirs.GNIRSConstants
+import edu.gemini.spModel.gemini.gsaoi.Gsaoi
+import edu.gemini.spModel.gemini.michelle.{InstMichelle, MichelleParams}
+import edu.gemini.spModel.gemini.nifs.InstNIFS
+import edu.gemini.spModel.gemini.niri.{InstNIRI, Niri}
+import edu.gemini.spModel.gemini.trecs.{InstTReCS, TReCSParams}
+import jsky.app.ot.editor.seq.Keys._
+
+
+/**
+ * Unique configurations represent sets of observes in a sequence which are done with the same
+ * instrument configuration and exposure times.
+ */
+case class UniqueConfig(count: Int, labels: String, config: Config) {
+
+  /** Single exposure time is either the given exposure time or the time on source divided by the number of images. */
+  def singleExposureTime: Double = singleTime.getOrElse(totalTime.getOrElse(0.0) / count)
+
+  /** Total exposure time is either total time on source (TReCS/Michelle) or the single exposure time times the number of images. */
+  def totalExposureTime: Double = totalTime.getOrElse(singleTime.getOrElse(0.0) * count)
+
+  /** TReCS and Michelle have an optional total time on source. */
+  private val totalTime = Option(config.getItemValue(INST_TIME_ON_SRC_KEY)).map(_.toString.toDouble)
+
+  /** Instruments other than TReCS and Michelle have a defined exposure time per image. */
+  private val singleTime = Option(config.getItemValue(INST_EXP_TIME_KEY)).map(_.toString.toDouble)
```

Comment:
You're right, it's a double. I replaced ```_.toString.toDouble``` with ```_.asInstanceOf[Double]```. You could argue that's potentially more dangerous(?). I guess it's "pick your poison", at least like that the double doesn't get converted to a string and back.